test/integration/src/peggy2/test_eth_transfers.py

```diff
@@ -140,14 +144,276 @@ def transfer_erc20_to_sifnode_and_back(ctx, token_sc, token_decimals, number_of_
         assert eth_balance_after_1 == send_amount_1 * (i + 1)
 
 
+# Lock an eth to
+# We expect the transfer from evm to sifchain to be successful,
+# We expect the transfer from sifchain to evm to fail, w/
+#   - Tokens burned on sifchain side
+#   - Tokens not depsoited on evm side
+#   - It SHOULD NOT halt the bridge
+def test_failhard_token_to_sifnode_and_back(ctx: EnvCtx):
+    test_eth_acct = ctx.create_and_fund_eth_account(fund_amount=fund_amount_eth)
+    test_sif_account = ctx.create_sifchain_addr(fund_amounts=[[fund_amount_sif, "rowan"]])
+                                                            #   [fund_amount_ceth_cross_chain_fee, ctx.ceth_symbol]])
+    test_account_token_balance = 30000
+
+    token_sc = deploy_failhard_for_test(ctx, test_eth_acct, test_account_token_balance)
+    token_addr = token_sc.address
+    sif_denom_hash = sifchain.sifchain_denom_hash(ctx.ethereum_network_descriptor, token_sc.address)
+
+    ctx.bridge_bank_lock_eth(test_eth_acct, test_sif_account, 5000)
+    ctx.advance_blocks(100)
+
+    sleep(10)
+    sif_balance_before = ctx.get_sifchain_balance(test_sif_account)
+    eth_token_balance_before = ctx.get_erc20_token_balance(token_addr, test_eth_acct)
+    # Locking erc20 token to sifchain
+    ctx.bridge_bank_lock_erc20(token_sc, test_eth_acct, test_sif_account, test_account_token_balance)
+    ctx.advance_blocks()
+
+    # Group these into 1 func
+    sif_balance_after = ctx.wait_for_sif_balance_change(test_sif_account, sif_balance_before)
+    sif_balance_delta = sifchain.balance_delta(sif_balance_before, sif_balance_after)
+    assert len(sif_balance_delta) == 1, "User should only have changes in token balance. Received {}".format(sif_balance_delta)
+    assert sif_denom_hash in sif_balance_delta, "User should see changes in the bridged token"
+    assert sif_balance_delta[sif_denom_hash] == test_account_token_balance
+
+    eth_token_balance_after = ctx.wait_for_eth_balance_change(test_eth_acct, eth_token_balance_before, token_addr=token_addr)
+    eth_token_balance_delta = eth_token_balance_after - eth_token_balance_before
+    assert eth_token_balance_delta == (test_account_token_balance * -1), "User's token on evm should have decreased by sent amount"
+
+    # The user has successfully locked token on evm, and got balance on sifchain
+    print("We have bridged the erc20 token into sif account and verified all account balances are as expected")
+
+    test_send_amount_back = test_account_token_balance - 15
```

Comment:
What is this magic number 15 for? Is it arbitrary to just verify a balance change? Please comment what it is.